ALBS-32: Groups of actions for pes.almalinux

soksanichenko opened the pull request:
- Some additional fixes to migration from SQLite to PostgreSQL
- The separate dockerfile for container with PostgreSQL
- Full implementation groups of actions

Commit:
ALBS-32: Groups of actions for pes.almalinux

- Some additional fixes to migration from SQLite to PostgreSQL
- The separate dockerfile for container with PostgreSQL
- Full implementation groups of actions

Korulag commented on ci/ansible/roles/deploy/tasks/migrate_sqlite_to_postgresql.yml:
```diff
@@ -5,23 +5,19 @@
-  file:
-    path: "{{ config_root }}/migration_scenario"
-    state: absent
+    - name: Run pqloader
```
Typo: `pqloader` -> `pgloader`

gost-serb commented on src/backend/api/handlers.py:
```diff
@@ -4,29 +4,57 @@
 from itertools import zip_longest
 from typing import (
     List,
-    Optional, Union, Tuple, Dict,
+    Optional,
+    Tuple,
```
Starting from 3.9 it's preferred to use built-in types (like `list`, `tuple`...) over the ones from typing module

gost-serb commented on src/backend/api/handlers.py:
```diff
@@ -345,6 +417,7 @@ def add_or_edit_action_handler(
         page: int,
+        group_id: int,
         page_size: int = PAGE_SIZE,
 ) -> Tuple[List[ActionData], Page]:
```
```suggestion
) -> tuple[list[ActionData], Page]:
```

gost-serb commented on src/backend/api/handlers.py:
```diff
@@ -458,9 +539,10 @@ def get_groups_of_actions_handler(
 ) -> Tuple[List[GroupActionsData], Page]:
```
```suggestion
) -> tuple[list[GroupActionsData], Page]:
```

gost-serb commented on src/backend/api/handlers.py:
```diff
@@ -230,12 +298,12 @@ def add_or_edit_group_of_actions_handler(
     with session_scope() as db_session:
         if is_new:
-            GroupActions.create_from_dataclass(
+            Group.create_from_dataclass(
                 session=db_session,
                 group_actions_data=group_actions_data,
             )
         else:
-            GroupActions.update_from_dataclass(
+            Group.update_from_dataclass(
                 session=db_session,
                 group_actions_data=group_actions_data,
             )
```
```python
            method = Group.create_from_class if is_new else Group.update_from_dataclass
            method(
                session=db_session,
                group_actions_data=group_actions_data,
            )
```

gost-serb commented on src/backend/db/data_models.py:
```diff
@@ -150,12 +151,14 @@ class GroupActionsData(BaseData):
     name: str = None
     description: str = None
     github_org: GitHubOrgData = None
+    actions_ids: List[int] = None
```
```suggestion
    actions_ids: list[int] = None
```

gost-serb commented on src/backend/common/forms.py:
```diff
@@ -131,14 +164,23 @@ def to_dataclass(self) -> GroupActionsData:
+                github_id=self.org.data,
+            ),
+            actions_ids=[
+                int(action_id.strip()) for action_id in
+                self.actions_ids.data.split(',')
```
```suggestion
                int(action_id.strip())
                for action_id in self.actions_ids.data.split(',')
```

This formatting makes it easier to see where is the result part and where is the loop itself

gost-serb commented on src/backend/db/data_models.py:
```diff
@@ -301,24 +306,27 @@ class ActionData(BaseData):
     in_package_set: List[PackageData] = field(default_factory=list)
     out_package_set: List[PackageData] = field(default_factory=list)
     arches: List[str] = field(default_factory=list)
+    groups: List[GroupActionsData] = field(default_factory=list)
```
Here and in other places `typing.List` is deprecated ([PEP-585](https://peps.python.org/pep-0585/)) starting from 3.9 (if I'm not mistaken) and should be replaced with built-in `list`

anfimovdm reviewed (approved):
lgtm

maccelf reviewed (approved):
LGTM

Commit:
ALBS-32: Groups of actions for pes.almalinux

- Some additional fixes to migration from SQLite to PostgreSQL
- The separate dockerfile for container with PostgreSQL
- Full implementation groups of actions

Korulag commented on src/backend/api/handlers.py:
```diff
@@ -182,16 +245,16 @@ def bulk_upload_handler(json_dict: dict) -> None:
-            'release',
-            'in_packageset',
-            'out_packageset',
+                'initial_release',
```
Why additional indent?

anfimovdm reviewed (approved):
lgtm

Commit:
ALBS-32: Groups of actions for pes.almalinux

- Some additional fixes to migration from SQLite to PostgreSQL
- The separate dockerfile for container with PostgreSQL
- Full implementation groups of actions
